drivers/gpu/drm/i915/i915_active.c

```diff
@@ -130,6 +131,47 @@ static inline void debug_active_assert(struct i915_active *ref) { }
 
 #endif
 
+#if defined(__FreeBSD__)
+static struct llist_head linux_kmem_cache_async_list = LLIST_HEAD_INIT();
+static struct llist_head linux_kmem_cache_addr_async_list = LLIST_HEAD_INIT();
+
+static void
+linux_kmem_cache_free_async_fn(struct irq_work *irqw)
+{
+	struct llist_node *freed_cache, *freed_addr;
+
+	while ((freed_cache = llist_del_first(&linux_kmem_cache_async_list)) != NULL &&
+		(freed_addr = llist_del_first(&linux_kmem_cache_addr_async_list)) != NULL)
+		kmem_cache_free((struct linux_kmem_cache*) freed_cache, freed_addr);
+}
+static DEFINE_IRQ_WORK(linux_kmem_cache_free_async_work,
+	linux_kmem_cache_free_async_fn);
+
+static void
+linux_kmem_cache_free_async(struct linux_kmem_cache *c, void *m)
+{
+	if (m == NULL)
+		return;
+
+	llist_add((void*) c, &linux_kmem_cache_async_list);
+	llist_add(m, &linux_kmem_cache_addr_async_list);
```

Comment:
You must store both pointers in the allocation pointed by m

Reply:
Should I add `*c` and `*m` subsequently in the same list, or use a struct with the current values of `*c` and `*m`?

Reply:
I realized we only use `global.slab_cache` so I have modified the code to only use that. If needed, I can revert the commit.

Reply:
> Should I add *c and *m subsequently in the same list, or use a struct with the current values of *c and *m?

Two sequential atomic operations are not atomic as a whole. Use a struct.